qiskit_experiments/data_processing/data_action.py

```diff
@@ -0,0 +1,76 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Defines the steps that can be used to analyse data."""
+
+from abc import ABCMeta, abstractmethod
+from typing import Any
+
+
+class DataAction(metaclass=ABCMeta):
+    """
+    Abstract action done on measured data to process it. Each subclass of DataAction must
+    define the way it formats, validates and processes data.
+    """
+
+    def __init__(self, validate: bool = True):
+        """
+        Args:
+            validate: If set to False the DataAction will not validate its input.
+        """
+        self._validate = validate
+
+    @abstractmethod
+    def _process(self, datum: Any) -> Any:
+        """
+        Applies the data processing step to the data.
+
+        Args:
+            datum: A single item of data which will be processed.
+
+        Returns:
+            processed data: The data that has been processed.
+        """
+
+    @abstractmethod
+    def _format_data(self, datum: Any, validate: bool = True) -> Any:
+        """
```

Comment:
This function doesn't need the `validate` parameter, it can use `self._validate`

Reply:
Done here: 58ca872